src/processor/bugsnag_stackwalk_wrapper.cc

```diff
@@ -40,6 +59,104 @@ int getErrorReportingThreadIndex(const ProcessState& process_state) {
   return index;
 }
 
+void destroyStackframe(void* self) {
+  Stackframe* stackframe = (Stackframe*)self;
+  if (NULL == stackframe) return;
+
+  freeAndInvalidate((void *)stackframe->filename);
+  freeAndInvalidate((void *)stackframe->method);
+  freeAndInvalidate((void *)stackframe->frameAddress);
+  freeAndInvalidate((void *)stackframe->loadAddress);
+  freeAndInvalidate((void *)stackframe->moduleId);
+  freeAndInvalidate((void *)stackframe->moduleName);
+  freeAndInvalidate((void *)stackframe->returnAddress);
+  freeAndInvalidate((void *)stackframe->symbolAddress);
+  freeAndInvalidate((void *)stackframe->codeFile);
+  freeAndInvalidate((void *)stackframe->trust);
+}
+
+void destroyStacktrace(void* self) {
+  Stacktrace* stacktrace = (Stacktrace*)self;
+  if (NULL == stacktrace) return;
+
+  for (int i = 0; i < stacktrace->frameCount; ++i) {
+    stacktrace->frames[i].destroy(&stacktrace->frames[i]);
```

Comment:
Is there any advantage in storing the `destroy` function on each struct? Now that we have moved the destroy logic out of the header it may be simpler to just call the specific destroy functions in each dedicated function (i.e. call `destroyStackframe(&stacktrace->frames[i])` here)

Reply:
Agreed, it doesn't add anything now, and your suggestion simplifies the code.